### ISSUE TITLE
improve error message to indicate better that duplicate map key is encountered

### ISSUE THREAD

**Issue by cppforlife** [k14s/ytt#69]
```
Error:
- __ytt_tpl3_start_node: disallowed to override key 'username' value
    in additional_db_values
      config.yml:35 |   username: #@ data.values.cc.db.password
    in <toplevel>
      config.yml:44 | --- #@ template.replace(db_config(merge_values(data.values.db, additional_db_values())))
```

**Comment by cppforlife** [comment]
changed to `expected key 'test1' to not be specified again (unless 'yaml/map-key-override' annotation is added)` in https://github.com/k14s/ytt/commit/84ec13bb5d9e6407eca124c5c9fc556d71dcb2b4

**Comment by cppforlife** [comment]
was shipped as part of 0.24.0
